src/main/java/com/example/springboot/dao/BookDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+package com.example.springboot.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Repository;
+
+import com.example.springboot.base.Book;
+import com.mysql.jdbc.Statement;
+
+/**
+ * This class implement methods from BookDAO interface
+ * @author Dmytro Kohut
+ * @version 1.1
+ */
+@Repository("bookDAO")
+public class BookDAOImpl implements DAOInterface<Book> {
```

Comment:
I don't like when services calls `Impl`. Rewrite it BookDAOImpl have to be `BookDAOService` and `DAOInterface` -> `IBookDAOService`